internal/engine/upper.go

```diff
@@ -150,6 +162,8 @@ func (u Upper) CreateManifests(ctx context.Context, manifests []model.Manifest,
 			handlePodEvent(ctx, engineState, pod)
 		case err := <-sw.errs:
 			return err
+		case <-time.After(refreshInterval):
```

Comment:
i don't understand what this code is doing. why would we need to do a hud update if nothing has changed?

Reply:
ok

i have a long-term concern that the design pattern here (have a central state object that's updated in a loop and sent to listeners) is basically isomorphic to the redux design pattern, and that we're going to end up reinventing a lot of the concepts there. So I'd rather borrow liberally rather than try inventing approaches that don't work.

The "show the age of a thing" is pretty much in every React 101 tutorial, and tends to be handled by component-local state, and tends to be discouraged in global state.